### ISSUE TITLE
System.ArgumentNullException: MonitorItem construtor, parameter deviceInstanceId

### ISSUE THREAD

**Issue by schittli** [emoacht/Monitorian#205]
Good morning

#### thank you very much for sharing your great work!
I would have bought a license long ago if the price was mentioned transparently and it didn't have to be bought via the M$ Store.

### I've got two exceptions

**I'm using Monitorian 2.16.0.0** 
_(I thought Monitorian would update itself or notify me when there was an update, so I never checked to see if there was a new version. I have just installed 3.0.1.)_

See: [exception.log](https://github.com/emoacht/Monitorian/files/6898526/exception.log)

I use a USB C Dockingstation with an assigned Monitor which is powered off during the night, while the notebook is kept running.
I received these exceptions for the first time, although I have had the docking station with the monitor for a long time and turn it off every night and on again in the morning.

#### I suspect that the first exception happened:
when I **turned off** the docking station (`[Date: 29.07.2021 00:40:58 Ver: 2.16.0.0]`):

```
[Date: 29.07.2021 00:40:58 Ver: 2.16.0.0]
System.ArgumentNullException: Value cannot be null.
Parameter name: deviceInstanceId HResult: -2147467261 ParamName: deviceInstanceId
   at Monitorian.Core.Models.Monitor.MonitorItem..ctor(String deviceInstanceId, String description, Byte displayIndex, Byte monitorIndex, Rect monitorRect, Boolean isReachable)
   at Monitorian.Core.Models.Monitor.UnreachableMonitorItem..ctor(String deviceInstanceId, String description, Byte displayIndex, Byte monitorIndex, Boolean isInternal)
   at Monitorian.Core.Models.Monitor.MonitorManager.<EnumerateMonitors>d__4.MoveNext()
   at Monitorian.Core.AppControllerCore.<<ScanAsync>b__46_0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at Monitorian.Core.AppControllerCore.<ScanAsync>d__46.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Monitorian.Core.AppControllerCore.<OnMonitorsChangeInferred>d__33.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.<>c.<ThrowAsync>b__6_0(Object state)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
```

#### I suspect that the 2nd exception happened:
when I **turned on** the docking station (`[Date: 29.07.2021 08:56:33 Ver: 2.16.0.0]`):

```
[Date: 29.07.2021 08:56:33 Ver: 2.16.0.0]
System.ArgumentNullException: Value cannot be null.
Parameter name: deviceInstanceId HResult: -2147467261 ParamName: deviceInstanceId
   at Monitorian.Core.Models.Monitor.MonitorItem..ctor(String deviceInstanceId, String description, Byte displayIndex, Byte monitorIndex, Rect monitorRect, Boolean isReachable)
   at Monitorian.Core.Models.Monitor.UnreachableMonitorItem..ctor(String deviceInstanceId, String description, Byte displayIndex, Byte monitorIndex, Boolean isInternal)
   at Monitorian.Core.Models.Monitor.MonitorManager.<EnumerateMonitors>d__4.MoveNext()
   at Monitorian.Core.AppControllerCore.<<ScanAsync>b__46_0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at Monitorian.Core.AppControllerCore.<ScanAsync>d__46.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Monitorian.Core.AppControllerCore.<OnMonitorsChangeInferred>d__33.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.<>c.<ThrowAsync>b__6_0(Object state)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.DispatcherOperation.InvokeImpl()
   at System.Windows.Threading.DispatcherOperation.InvokeInSecurityContext(Object state)
   at MS.Internal.CulturePreservingExecutionContext.CallbackWrapper(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at MS.Internal.CulturePreservingExecutionContext.Run(CulturePreservingExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Windows.Threading.DispatcherOperation.Invoke()
   at System.Windows.Threading.Dispatcher.ProcessQueue()
   at System.Windows.Threading.Dispatcher.WndProcHook(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs)
   at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam)
   at MS.Win32.UnsafeNativeMethods.DispatchMessage(MSG& msg)
   at System.Windows.Threading.Dispatcher.PushFrameImpl(DispatcherFrame frame)
   at System.Windows.Threading.Dispatcher.PushFrame(DispatcherFrame frame)
   at System.Windows.Application.RunDispatcher(Object ignore)
   at System.Windows.Application.RunInternal(Window window)
   at System.Windows.Application.Run(Window window)
   at Monitorian.App.Main()
```

Kind regards,
Thomas

**Comment by emoacht** [comment]
Thanks for the report.
It reminds me of any odd thing can happen when docking or undocking. I can suppress that exceptions.

**Comment by emoacht** [comment]
Fixed by 416b18efc18401ff83d3d6bcdd48ac600dc33d5c
